src/rpc/mining.cpp

```diff
@@ -157,6 +158,12 @@ UniValue generateBlocks(boost::shared_ptr<CReserveScript> coinbaseScript, int nG
             coinbaseScript->KeepScript();
         }
     }
+
+    CValidationState state;
+    LOCK(cs_main);
+    FlushStateToDisk(state, FLUSH_STATE_ALWAYS); // we made lots of blocks
+    CBlockIndex *pindexNewTip = chainActive.Tip();
+    uiInterface.NotifyBlockTip(false, pindexNewTip);
     return blockHashes;
```

Comment:
The call to uiInterface is redundant, the ui is updated after each block is connected, at the end of ActivateBestChainStep.  It's somewhat minor but I also wonder whether this section should be moved up into the loop, and executed right after ProcessNewBlock() is called?

Reply:
I thought so too, but when I tested it, it showed 1 additional block no matter how many I mined in the GUI.   I realized that there is a time wrapper around the code in BlockTipChanged (clientmodel.cpp) that ensures that the GUI is not updated too often.

By notifying with the the first arg "false" I avoid this time wrapper and force the update once all blocks are mined.

Reply:
make sense